src/Shared/src/CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -317,7 +320,7 @@ public int Execute(params string[] args)
                     option = null;
                 }
 
-                if (!processed && argumentsAssigned)
+                if (!processed && !argumentsAssigned)
```

Comment:
What's the reason this was inverted?

Reply:
I named this variable wrong. `argumentsAssigned` was true when no variable was set. Although everything was working right, I just flip everything to make the name right and hence invert this check!

Reply:
Ah, I missed that you added this boolean because I was only looking at the most recent commit ;). Thanks for the clarification!